doc/design/draft/named-segments.md

```diff
@@ -0,0 +1,219 @@
+    1. Iterate over all process `userGroups`
+    2. Determine if the user group name matches the name of one of the segments
+    3. If it does:
+        1. If a match has already been previously made, return an error indicating that the publisher must only have write access to one segment
+        2. If not, verify the process has access to the segment and record the match
+    4. At the end of iteration, if a matching segment has been found, return the segment information
+    5. Otherwise return an error indicating that no matching segment has been found
+
+#### Runtime requesting segments to map
+
+In the client process, while initializing the runtime, given:
+* `userGroups` - A list of POSIX user groups a publishing process belongs to called 
+* `segmentContainer` - A list of shared memory segment information
+* `premappedSegments` - An enum indicating which group of segments should be mapped by default
+* `segmentFilter` - A (possibly empty) list of segment names to filter against
+
+In order to determine which segments to map:
+
+1. Map segments according to the `premappedSegments` option
+  1. If read access segments are requested, map every segment the user has read access to, but not write access
+  2. If write access segments are requested, map every segment the user has write access to
+  3. If read and write access segmeents are requested, map every segment the user has either read or write access to
+  4. If no segments are requested, do nothing
+2. If `segmentFilter` is not empty
+    1. Iterate over each name in the filter
+    2. If there is a segment that matches the name in the filter
+        1. If the segment has already been mapped, return an error indicating that a segment has been requested twice. The error should contain access permissions and the value of the `premappedSegments` option the user understands why it has already been mapped.
+        2. Otherwise, if the process has access rights to that segment, map it
+        2. If the process does not have access rights, return an error indicating that a segment was requested the runtime does not have access to.
+    3. If there is no segment that matches the name in the filter, return an error indicating that there is no segment matching the one requested to be mapped.
+
+## Development Roadmap
+
+- [ ] Extend the RouDi config to support specifying segment names to be added to the `SegmentConfig`
+- [ ] Add the name to the `MePooSegment` data structure and allow multiple write-access segments to be mapped.
+- [ ] Update producer options structs to include segment names
+- [ ] Update producer segment selection logic to take the segment name into account
+
+### Optional
+
+- [ ] Refactor runtime initialization to use builder pattern
+- [ ] Add segment filter and apply it to segment mapping during runtime initialization. 
+- [ ] Add flag to specify whether endpoints requesting non-mapped segments should fail or whether segments should be created dynamically
```

Comment:
I think here some further discussion are needed :)

Reply:
Yeah. It sounds like most of the existential questions revolve around refactoring the runtime. I wonder if it would make sense to move all this optional section to a different design draft and just focus on allowing publishers to specify segments in this design. The feature doesn't actually require any changes to the runtime API - just slight tweaks to the segment matching algorithms for publishers.

Reply:
I might be biased due to the fact that, the segment naming is the concrete feature I need to deliver to my company to achieve our goals. But more generally I find it good software engineering practice to separate concerns as much as possible and deliver features in bite-sized pieces 😉

Reply:
Sure. Go ahead. It basically all boils down to create an API which does not impose a high burden on maintenance. As you know, the community is not that big and getting a lot of bug reports because of [Hyrum's Law](https://www.hyrumslaw.com/) is something we want to avoid.